sled-agent/src/rack_setup/service.rs

```diff
@@ -130,8 +131,27 @@ use std::iter;
 use std::net::{Ipv6Addr, SocketAddrV6};
 use std::time::Duration;
 use thiserror::Error;
+use tokio::sync::watch;
 use uuid::Uuid;
 
+/// For tracking the current RSS step and sending notifications about it.
+pub struct RssProgress {
+    step: RssStep,
+    step_tx: watch::Sender<RssStep>,
+}
+
+impl RssProgress {
+    pub fn new(step_tx: watch::Sender<RssStep>) -> Self {
+        let _ = step_tx.send(RssStep::Starting);
```

Comment:
If `let _ = ` to ignore the error is annoying, a couple alternatives:

```rust
step_tx.send_modify(|step| *step = RssStep::Starting);
step_tx.send_replace(RssStep::Starting);
```

Both of these perform unconditional updates regardless of the number of receivers, so they're infallible. 

Between this and the previous comment, I'm tempted to suggest deleting `RssProgress` and replacing all the `rss_step.update(foo)` calls with `rss_step_tx.send_replace(foo)`, but if you think `RssProgress` might gain other stuff or is just nicer to look at, I wouldn't object.

Reply:
I had plans...

```
warning: field `step` is never read
   --> sled-agent/src/bootstrap/rack_ops.rs:300:9
    |
294 |     Initializing {
    |     ------------ field in this variant
...
300 |         step: RssStep,
    |         ^^^^
    |
    = note: `#[warn(dead_code)]` on by default
```

But, now the compiler agrees with you...

Reply:
I had no choice but to take your advice.